File: frontend/src/app/features/pedidos/domain/interfaces/repositories/IProveedorRepository.ts
import { Proveedor } from '../../entities/proveedor';

/**
 * Contrato del repositorio de proveedores.
 * @remarks Implementado por {@link ProveedorRepository}.
 */
export interface IProveedorRepository {
  /** Devuelve todos los proveedores (activos e inactivos). */
  getAll(): Promise<Proveedor[]>;
  /** Devuelve únicamente los proveedores activos. */
  getActivos(): Promise<Proveedor[]>;
  /**
   * Obtiene un proveedor por su identificador.
   * @param id Identificador del proveedor.
   * @returns Entidad {@link Proveedor} o `null` si no existe.
   */
  getById(id: number): Promise<Proveedor | null>;
}

File: backend/ERP.Backend/Domain/Interfaces/IRepository/IProveedorRepository.cs
using Domain.Entities;

namespace Domain.Interfaces
{
    /// <summary>
    /// Interfaz del repositorio del maestro de proveedores.
    /// </summary>
    public interface IProveedorRepository
    {
        /// <summary>
        /// Obtiene todos los proveedores del sistema.
        /// </summary>
        /// <returns>Lista de <see cref="Proveedor"/> con todos los registros.</returns>
        Task<List<Proveedor>> GetAllAsync();

        /// <summary>
        /// Obtiene un proveedor por su identificador único.
        /// </summary>
        /// <param name="proveedorID">Identificador único del proveedor a buscar.</param>
        /// <returns>El <see cref="Proveedor"/> encontrado, o <c>null</c> si no existe.</returns>
        Task<Proveedor?> GetByIdAsync(int proveedorID);
    }
}